# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.cpp
# Commit: ff0876ea84c9
# Full Hash: ff0876ea84c9326d63063fdaba470b521ef10e23
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Enable VideoDecoder on Linux only r=padenot
#   
#   The codec support will be done in Bug 1840508 later
#   
#   Depends on D182291
# ==============================================================================

diff -r 8fe486442687 -r ff0876ea84c9 dom/media/webcodecs/VideoDecoder.cpp
--- a/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 22:33:14 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 22:33:14 2023 +0000
@@ -147,8 +147,21 @@
   return types;
 }
 
+static bool IsOnLinux() {
+#if defined(XP_LINUX) && !defined(ANDROID)
+  return true;
+#else
+  return false;
+#endif
+}
+
 // https://w3c.github.io/webcodecs/#check-configuration-support
 static bool CanDecode(const VideoDecoderConfig& aConfig) {
+  // Bug 1840508: H264-annexb doesn't work on non-linux platform. We only enable
+  // Linux for now.
+  if (!IsOnLinux()) {
+    return false;
+  }
   // TODO: Instead of calling CanHandleContainerType with the guessed the
   // containers, DecoderTraits should provide an API to tell if a codec is
   // decodable or not.